### ISSUE TITLE
`rename` does not work well when the renamed output is one of final outputs of workflow

### ISSUE THREAD

**Issue by ROCKTAKEY** [arunisaac/ccwl#8]

**Comment by ROCKTAKEY** [comment]
Sorry, I sent incomplete issue.

`rename` should renames also workflow output, but it renames nothing. For example, the script below:
```scheme
(define print
  (command #:inputs (message #:type string)
           #:run "echo" message
           #:outputs (printed-message #:type stdout)))

(workflow ((message1 #:type string)
           (message2 #:type string))
          (tee (pipe (print (print1) #:message message1)
                     (rename #:out1 printed-message))
               (print (print2) #:message message2)))
```
generates the `cwl` file below:
```cwl
cwlVersion:
  v1.2
class:
  Workflow
requirements:
  SubworkflowFeatureRequirement: {}
inputs:
  message1:
    type:
      string
  message2:
    type:
      string
outputs:
  "printed-message":
    type:
      File
    outputSource:
      "print1/printed-message"
  "printed-message":
    type:
      File
    outputSource:
      "print2/printed-message"
steps:
  print1:
    in:
      message:
        message1
    out:
      - "printed-message"
    run:
      cwlVersion:
        v1.2
      class:
        CommandLineTool
      arguments:
        - position:
            0
          valueFrom:
            echo
      inputs:
        message:
          type:
            string
          inputBinding:
            position:
              1
      outputs:
        "printed-message":
          type:
            stdout
  print2:
    in:
      message:
        message2
    out:
      - "printed-message"
    run:
      cwlVersion:
        v1.2
      class:
        CommandLineTool
      arguments:
        - position:
            0
          valueFrom:
            echo
      inputs:
        message:
          type:
            string
          inputBinding:
            position:
              1
      outputs:
        "printed-message":
          type:
            stdout
```
which is not including any string "out1" and where `output.printed-message` is duplicated.

so when running `cwltool aa.cwl --message1 "a" --message2 "b"`, it causes error:
```
ERROR I'm sorry, I couldn't load this CWL file, try again with --debug for more information.
The error was: while constructing a mapping
  in "file:///home/sandbox/sandbox/ccwl/aa.cwl", line 15, column 3
found duplicate key "printed-message" with value "ordereddict([('type', 'File'), ('outputSource', 'print2/printed-message')])" (original value: "ordereddict([('type', 'File'), ('outputSource', 'print1/printed-message')])")
  in "file:///home/sandbox/sandbox/ccwl/aa.cwl", line 20, column 3

To suppress this check see:
    http://yaml.readthedocs.io/en/latest/api.html#duplicate-keys

Duplicate keys will become an error in future releases, and are errors
by default when using the new API.
```

**Comment by arunisaac** [comment]
Good catch!

I have now fixed this in 05f8c9b109146a5647a31cfe4c55661f47ff75f0. Could
you check?

Thanks!

**Comment by ROCKTAKEY** [comment]
It works well! Thanks a lot!
